commands/post_test.go

```diff
@@ -0,0 +1,106 @@
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestPostCreateCmdF() {
+	s.Run("create a post with empty text", func() {
+		cmd := &cobra.Command{}
+
+		err := postCreateCmdF(s.client, cmd, []string{"some-channel", ""})
+		s.Require().NotNil(err)
+	})
+
+	s.Run("no channel specified", func() {
+		msgArg := "some text"
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("message", msgArg, "")
+
+		err := postCreateCmdF(s.client, cmd, []string{"", msgArg})
+		s.Require().NotNil(err)
```

Comment:
To distinguish between the different errors, we can use `EqualError` here:

```suggestion
		s.Require().EqualError(err, "Message cannot be empty")
```